lib/pf/Switch/Trapeze/WLC.pm

```diff
@@ -0,0 +1,139 @@
+package pf::Switch::Trapeze::WLC;
+
+=head1 NAME
+
+pf::Switch::Trapeze::WLC
+
+=head1 SYNOPSIS
+
+Module to manage Juniper (Trapeze) controllers
+
+=cut
+
+use strict;
+use warnings;
+
+use Log::Log4perl;
+use Net::Appliance::Session;
+use POSIX;
+
+use base ('pf::Switch::Trapeze');
+
+use pf::config;
+# importing switch constants
+use pf::Switch::constants;
+use pf::util;
+use pf::constants;
+
+#sub supportsRoleBasedEnforcement { return $TRUE; }
+=head1 STATUS
+
+=head1 BUGS AND LIMITATIONS
+
+=over
+
+=item returnRadiusAccessAccept
+
+Prepares the RADIUS Access-Accept reponse for the network device.
+
+Default implementation.
+
+=cut
+sub returnRadiusAccessAccept {
+    my ($self, $args) = @_;
+    my $logger = $self->logger();
+
+    my $radius_reply_ref = {};
+    use Data::Dumper;
+
+    # should this node be kicked out?
+    my $kick = $self->handleRadiusDeny($args);
+    return $kick if (defined($kick));
+
+    # Inline Vs. VLAN enforcement
+    my $role = "";
+    if ( (!$args->{'wasInline'} || ($args->{'wasInline'} && $args->{'vlan'} != 0) ) && isenabled($self->{_VlanMap})) {
+        if(defined($args->{'vlan'}) && $args->{'vlan'} ne "" && $args->{'vlan'} ne 0){
+            if ( $args->{'vlan'} =~ /^[-\d]+$/ ) {
+                $radius_reply_ref = {
+                    'Tunnel-Medium-Type' => $RADIUS::ETHERNET,
+                    'Tunnel-Type' => $RADIUS::VLAN,
+                    'Tunnel-Private-Group-ID' => $args->{'vlan'},
+                };
+            } 
+            else {
+                $radius_reply_ref = {
+                    'Trapeze-VLAN-Name'  => $args->{'vlan'},
+                };
+            }
+            $logger->info("(".$self->{'_id'}.") Added VLAN $args->{'vlan'} to the returned RADIUS Access-Accept");
+        }
+        else {
+            $logger->debug("(".$self->{'_id'}.") Received undefined VLAN. No VLAN added to RADIUS Access-Accept");
+        }
+    }
+
+    if ( isenabled($self->{_RoleMap}) && $self->supportsRoleBasedEnforcement()) {
+        $logger->debug("Network device (".$self->{'_id'}.") supports roles. Evaluating role to be returned");
+        if ( defined($args->{'user_role'}) && $args->{'user_role'} ne "" ) {
+            $role = $self->getRoleByName($args->{'user_role'});
+        }
+        if ( defined($role) && $role ne "" ) {
+            $radius_reply_ref = {
+                %$radius_reply_ref,
+                $self->returnRoleAttributes($role),
+            };
+            $logger->info('role answer' . Dumper($radius_reply_ref));
```

Comment:
Useless since if $radius_reply_ref is not empty we already know it by the log before in the code.